### ISSUE TITLE
Build instructions for linux needed

### ISSUE THREAD

**Issue by tavasti** [debauchee/barrier#28]
Build instructions for linux are needed. Plain cmake + make does not work. Make fails on 
/home/tavasti/Downloads/barrier-2.0.0/src/./lib/common/common.h:27:3: error: #error "config.h missing"
 # error "config.h missing"

**Comment by walker0643** [comment]
I believe I saw you getting help with this on IRC, @tavasti ... did everything work out?

And, yes, documentation is lacking at this point.

**Comment by tavasti** [comment]
Yes, I got things rolling. Point was run clean_build.sh and if you are not on git version, in cmake/Version.cmake adding line like:
 set (BARRIER_REVISION 12345678)

**Comment by yupi2** [comment]
You might be able to do `cmake -DBARRIER_REVISION 123123123 .....etc`

**Comment by VertigoRay** [comment]
Just walked through it on a new Linux Mint installation:

```bash
# Clone Repo
git clone https://github.com/debauchee/barrier.git
cd barrier

# Install Dependencies
sudo apt -y install cmake gcc build-essential libx11-dev libavahi-compat-libssdnssd-dev libxtst-dev qtbase5-dev libssl-dev

# Build Barrier
./clean_build.sh

# Run Setup
./build/bin/barrier
```

**Comment by Persei08** [comment]
Thx @VertigoRay , for me on ubuntu 18.04, it was also necessary to install `libcurl4-nss-dev` (and there is a typo for libavahi-compat-libdnssd-dev)

Thanks everyone for this project

**Comment by VertigoRay** [comment]
I'm glad it helped. 😊 Fixed the typo.

**Comment by tavasti** [comment]
For release-versions (tar or zip) instructions need bit more fidling, and those should be also covered.

**Comment by bezoris** [comment]
> # Install Dependencies
> sudo apt -y install cmake gcc build-essential libx11-dev libavahi-compat-libdnssd-dev libxtst-dev qtbase5-dev libssl-dev
> 
> # Build Barrier
> ./clean_build.sh
> 
> # Run Barrier
> ./build/bin/barrier
> ```

Many thanks! 
Successful build on Debian stretch after installing **libcurl3-nss** separately followed by **libcurl4-nss-dev**; already had libcurl3 and libcurl3-gnutls installed.

Was going a bit batty trying to figure this out. Thanks again!

**Comment by adeperio** [comment]
Can confirm @dayne 's instructions worked for me on ubuntu 18
